### PR TITLE
fix(engine) Jump out of the loop when err is nil

### LINKED ISSUE #19841: The createsnapshot() of backup() will be executed repeatedly when err is nil
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Analysis code Engine.Backup()

__Expected behavior:__
Jump out of the loop when err is nil

__Actual behavior:__
no break when err is nil

__Environment info:__

* System info: Run `uname -srm` and copy the output here
* InfluxDB version: Run `influxd version` and copy the output here
* Other relevant environment details: Container runtime, disk info, etc

__Config:__
Copy any non-default config values here or attach the full config as a gist or file.

__Logs:__
Include snippet of errors in log.

__Performance:__
Generate profiles with the following commands for bugs related to performance, locking, out of memory (OOM), etc.

```sh
# Commands should be run when the bug is actively happening.
# Note: This command will run for at least 30 seconds.
curl -o profiles.tar.gz "http://localhost:8086/debug/pprof/all?cpu=true"
curl -o vars.txt "http://localhost:8086/debug/vars"
iostat -xd 1 30 > iostat.txt
# Attach the `profiles.tar.gz`, `vars.txt`, and `iostat.txt` output files.
```

### PR DESCRIPTION
**Pull request by zddr**
close #19841 
Add judgment logic

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)
- [x] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by zddr** [comment]
@stuartcarnie

File: tsdb/engine/tsm1/engine.go [hunk 1]
```diff
@@ -917,7 +917,9 @@ func (e *Engine) Backup(w io.Writer, basePath string, since time.Time) error {
 	var path string
 	for i := 0; i < 3; i++ {
 		path, err = e.CreateSnapshot()
-		if err != nil {
+		if err == nil {
+			break
+		} else if err != nil {
```
Comment (stuartcarnie) [inline tsdb/engine/tsm1/engine.go:922]: Thanks for the improvement!

The `else if err != nil` is unnecessary with your change and possibly clearer as:

```go
switch {
case err == nil:
	break LOOP
case errors.Is(err, ErrSnapshotInProgress):
	backoff := time.Duration(math.Pow(32, float64(i))) * time.Millisecond
	time.Sleep(backoff)
default:
	return err
}
```

**Review by stuartcarnie — CHANGES_REQUESTED** [review]
Great catch, however, @benbjohnson might want to look, as he may be actively working in this area.

File: tsdb/engine/tsm1/engine.go (hunk 1 above)
Comment (zddr) [reply tsdb/engine/tsm1/engine.go]: Good idea. I've changed it

**Comment by lesam** [comment]
Fixed a while ago in https://github.com/influxdata/influxdb/pull/20527, closing this one.

**Pull request closed without merging**